src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -1125,10 +1125,29 @@ trait ChoosesMode {
           if (isFirstOrLastTrip) WALK_BASED.allowedBeamModesForFirstAndLastLeg
           else WALK_BASED.allowedBeamModes
         walkBasedModes ++ enabledModes
+      case Some(BIKE_BASED) =>
+        if (BIKE_BASED.vehicleSharedOrNot(availablePersonalStreetVehicles)) {Seq[BeamMode]()}
+        else BIKE_BASED.allowedBeamModes
```

Comment:
So let's say the person has access to both a shared bike and a household bike -- if they are on a bike based tour we'd still want them to have access to the BIKE mode, but we'd want them to only be able to use the personal bike for that bike trip. Which makes me think that we might not want to make the change here but instead below

Reply:
True

Reply:
So to clarify -- we want to return an empty sequence of possible modes only if all available bikes are shared (rather than if any are)

Reply:
Yes, I agree!